src/modules/cc.rs

```diff
@@ -0,0 +1,125 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::c::CConfig;
+use crate::configs::cpp::CppConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+
+use crate::segment::Segment;
+
+use semver::Version;
+use std::borrow::Cow;
+use std::ops::Deref;
+use std::sync::LazyLock;
+
+pub enum Lang {
+    C,
+    Cpp,
+}
+
+fn is_cc_project(config: &CConfig, context: &Context) -> Option<bool> {
+    Some(
+        context
+            .try_begin_scan()?
+            .set_extensions(&config.detect_extensions)
+            .set_files(&config.detect_files)
+            .set_folders(&config.detect_folders)
+            .is_match(),
+    )
+}
+
+fn parse_module(
+    context: &Context,
+    module: &mut Module,
+    config: CConfig,
+    compilers: [(&str, &str); 2],
+) -> Result<Vec<Segment>, crate::formatter::string_formatter::StringFormatterError> {
+    StringFormatter::new(config.format).and_then(|formatter| {
+        let cc_compiler_info = LazyLock::new(|| context.exec_cmds_return_first(config.commands));
+
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "name" => {
+                    let cc_compiler_info = &cc_compiler_info.deref().as_ref()?.stdout;
+
+                    let cc_compiler = 'search: {
+                        for (compiler_name, compiler_hint) in compilers {
+                            if cc_compiler_info.contains(compiler_hint) {
+                                break 'search Some(compiler_name);
+                            }
+                        }
+                        None
+                    };
```

Comment:
I would prefer `find_map` or a combination of `find` and `map` here.
E.g. (untested):
```suggestion
                    let cc_compiler = compilers.iter().find_map(|(compiler_name, compiler_hint)| {
                        cc_compiler_info.contains(compiler_hint).then_some(compiler_name)
                    };
```

Reply:
done

Reply:
done